app/modules/listings/components/listing_element_actions.svelte

```diff
@@ -0,0 +1,109 @@
+<script>
+  import { clone } from 'underscore'
+  import { icon } from '#app/lib/icons'
+  import { moveArrayElement } from '#app/lib/utils'
+  import Spinner from '#general/components/spinner.svelte'
+  import { removeElement, updateElement } from '#listings/lib/listings'
+  import { i18n } from '#user/lib/i18n'
+
+  export let isReordering, element, paginatedElements, listingId, flash
+
+  let isLoading
+  const { _id, uri } = element
+  $: hasSeveralElements = paginatedElements.length > 1
+  $: index = paginatedElements.findIndex(obj => obj.uri === uri)
+
+  async function _removeElement () {
+    removeElement(listingId, uri)
+      .then(() => {
+        // Enhancement: after remove, have an "undo" button
+        paginatedElements.splice(index, 1)
+        paginatedElements = paginatedElements
+      })
+      .catch(err => flash = err)
+  }
+
+  async function reorder (newIndex) {
+    isReordering = isLoading = true
+    flash = null
+    const oldElements = clone(paginatedElements)
+    moveArrayElement(paginatedElements, index, newIndex)
+    // moveArrayElement(elements, index, newIndex)
+    paginatedElements = paginatedElements
```

Comment:
with `moveArrayElement` modified to return the array, as suggested above:
```suggestion
    paginatedElements = moveArrayElement(paginatedElements, index, newIndex)
```

Reply:
done in 0543120ca